### PR TITLE
[Python][Client] Fix delimiter collision issue #5981

### LINKED ISSUE #5981: [BUG][Python] Single quotes in default/example string cause invalid syntax error in generated test code
#### Bug Report Checklist

- [x] Have you provided a full/minimal spec to reproduce the issue?
- [x] Have you validated the input using an OpenAPI validator ([example](https://apidevtools.org/swagger-parser/online/))?
- [x] What's the version of OpenAPI Generator used?
- [x] Have you search for related issues/PRs?
- [x] What's the actual output vs expected output?
- [ ] [Optional] Bounty to sponsor the fix ([example](https://www.bountysource.com/issues/66123212-javascript-client-produces-a-wrong-object-for-a-string-enum-type-that-is-used-with-ref))

##### Description

Single quotes in default or example values of type string (e.g. 0xA2D2’C3B0’A2D2’C3B0) cause invalid syntax error in generated test code for Python. The error is a delimiter collision when a value containing single quote is delimited with the same quote. So in a string assignment in generated test code, the above value becomes ’0xA2D2’C3B0’A2D2’C3B0’ instead of “0xA2D2’C3B0’A2D2’C3B0”.

##### openapi-generator version

4.2.3, master

##### OpenAPI declaration file content or url
```yaml
openapi: 3.0.0
info:
  title: Single quote example
  version: 1.0.0
paths:
  /singleQuoteExample:
    post:
      summary: Test single quote in default and example values.
      requestBody:
        required: true
        content:
          application/json:
            schema:
              type: object
              properties:
                singleQuoteInDefault1:
                  type: string
                  default: Text containing 'single' quote
                singleQuoteInDefault2:
                  type: string
                  default: "8'998'999'998'000'000"
                singleQuoteInExample1:
                  type: string
                  example: "Text containing 'single' quote"
                singleQuoteInExample2:
                  type: string
                  example: 8'998'999'998'000'000
      responses:
        '201':
          description: OK
```
##### Command line used for generation

java -jar openapi-generator-cli.jar generate -g python -i example.yaml -o example-client

##### Steps to reproduce

Run the above command.

Expected output:
```
def make_instance(self, include_optional):
    if include_optional :
        return InlineObject(
            single_quote_in_default1 = "Text containing 'single' quote", 
            single_quote_in_default2 = "8'998'999'998'000'000", 
            single_quote_in_example1 = "Text containing 'single' quote", 
            single_quote_in_example2 = "8'998'999'998'000'000"
        )
    else :
        return InlineObject(
    )
```
Actual output:
```
def make_instance(self, include_optional):
    if include_optional :
        return InlineObject(
            single_quote_in_default1 = 'Text containing 'single' quote', 
            single_quote_in_default2 = '8'998'999'998'000'000', 
            single_quote_in_example1 = 'Text containing 'single' quote', 
            single_quote_in_example2 = '8'998'999'998'000'000'
        )
    else :
        return InlineObject(
    )
```

##### Related issues/PRs

#2808 #2809

##### Suggest a fix

- Wrap using double quotes to avoid the need to escape any embedded single quotes. The changes are in these 2 lines in `PythonClientCodegen.java`.

Line 683 (method `toDefaultValue`):
Current: 
`return "'" + ((String) p.getDefault()).replaceAll("'", "\'") + "'";`
Change to:
`return "\"" + p.getDefault() + "\"";`

Line 727 (method `toExampleValueRecursive`):
Current: 
`example = "'" + example + "'";`
Change to:
`example = "\"" + example + "\"";`

- Add/update test cases in `PythonClientCodegenTest.java`.

### PR DESCRIPTION
**Pull request by fullcircle23**
Wrapped the string literals for the default and example values using double quotes (instead of single quotes) to avoid the need to escape any embedded single quotes and also to avoid any invalid syntax errors due to delimiter collision in the generated test code.

Test Files:
- Updated 1 test.
- Added 1 test.

Updated sample files after running the following:
- bin/python-petstore-all.sh
- bin/openapi3/python-petstore.sh
- bin/openapi3/python-experimental-petstore.sh

### Open Issues

If merged, this will resolve #5981 

### PR checklist

- [x] Read the [contribution guidelines](https://github.com/openapitools/openapi-generator/blob/master/CONTRIBUTING.md).
- [ ] If contributing template-only or documentation-only changes which will change sample output, [build the project](https://github.com/OpenAPITools/openapi-generator#14---build-projects) before.
- [x] Run the shell script(s) under `./bin/` (or Windows batch scripts under`.\bin\windows`) to update Petstore samples related to your fix. This is important, as CI jobs will verify _all_ generator outputs of your HEAD commit, and these must match the expectations made by your contribution. You only need to run `./bin/{LANG}-petstore.sh`, `./bin/openapi3/{LANG}-petstore.sh` if updating the code or mustache templates for a language (`{LANG}`) (e.g. php, ruby, python, etc).
- [x] File the PR against the [correct branch](https://github.com/OpenAPITools/openapi-generator/wiki/Git-Branches): `master`, `4.3.x`, `5.0.x`. Default: `master`.
- [x] Copy the [technical committee](https://github.com/openapitools/openapi-generator/#62---openapi-generator-technical-committee) to review the pull request if your PR is targeting a particular programming language.

Note: This PR replaces PR #5982 

@spacether @wing328

### CODE REVIEW

**Comment by wing328** [comment]
The python (oas3) tests failed. Please refer to https://travis-ci.org/github/OpenAPITools/openapi-generator/builds/691670401 for more information.

**Comment by fullcircle23** [comment]
> The python (oas3) tests failed. Please refer to https://travis-ci.org/github/OpenAPITools/openapi-generator/builds/691670401 for more information.

Took a while to debug this. It seems I need to wrap the default and example "enum" property values as well.

File: modules/openapi-generator/src/main/java/org/openapitools/codegen/languages/PythonClientCodegen.java [hunk 1]
```diff
@@ -902,43 +907,45 @@ public void setParameterExampleValue(CodegenParameter p) {
...
             }
             example = "'" + escapeText(example) + "'";
-        } else if (!languageSpecificPrimitives.contains(type)) {
+          } else if (!languageSpecificPrimitives.contains(type)) {
             // type is a model class, e.g. User
             example = this.packageName + "." + type + "()";
-        } else {
+          } else {
             LOGGER.warn("Type " + type + " not handled properly in setParameterExampleValue");
+          }
```
Comment (wing328) [inline modules/openapi-generator/src/main/java/org/openapitools/codegen/languages/PythonClientCodegen.java:948]: FYI. Looks like these changes are not indented properly (2-space vs 4-space). I can fix it later after merging this PR.

Comment (fullcircle23) [reply modules/openapi-generator/src/main/java/org/openapitools/codegen/languages/PythonClientCodegen.java:948]: Noted. Thank you!

**Comment by wing328** [comment]
If no further feedback/question on this PR, I'll merge it over the weekend.

**Comment by wing328** [comment]
Unfortunately, we need to roll back this change as it's causing CircleCI failure after merging into master. Please refer to the PR with the details: https://github.com/OpenAPITools/openapi-generator/pull/6514

I'll see if I can file a PR to fix the issue in the coming weekend.

**Pull request merged**